subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2094,9 +2094,70 @@ void lwm2m_engine_get_queue_mode(char *queue)
 	}
 }
 
+static int lwm2m_engine_allocate_resource_instance(struct lwm2m_engine_res *res,
+						   struct lwm2m_engine_res_inst **res_inst,
+						   uint8_t resource_instance_id)
+{
+	int i;
+
+	if (!res->res_instances || res->res_inst_count == 0) {
+		return -ENOMEM;
+	}
+
+	for (i = 0; i < res->res_inst_count; i++) {
+		if (res->res_instances[i].res_inst_id ==
+		    RES_INSTANCE_NOT_CREATED) {
+			break;
+		}
+	}
+
+	if (i >= res->res_inst_count) {
+		return -ENOMEM;
+	}
+
+	res->res_instances[i].res_inst_id = resource_instance_id;
+	*res_inst = &res->res_instances[i];
+	return 0;
+}
+
+int lwm2m_engine_get_create_res_inst(struct lwm2m_obj_path *path, struct lwm2m_engine_res **res,
+				     struct lwm2m_engine_res_inst **res_inst)
+{
+	int ret;
+	struct lwm2m_engine_res *r = NULL;
+	struct lwm2m_engine_res_inst *r_i = NULL;
+
+	ret = path_to_objs(path, NULL, NULL, &r, &r_i);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (!r) {
+		LOG_ERR("resource %u not found", path->res_id);
```

Comment:
`resource 7 not found` in the serial log is not much information for the user to start debugging. It does not indicate who called, and what object tried to do what.

I would prefer to drop this error log, as we are returning proper error code, so let the upper layer to deal with it.

Reply:
Ok I remove.